test/recipes/30-test_evp_data/evpciph.txt

```diff
@@ -2509,32 +2509,43 @@ Ciphertext = 64a0861575861af460f062c79be643bd5e805cfd345cf389f108670ac76c8cb24c6
 
 Title = ZUC/EEA3 test vectors
 
+# The trailing zero bytes are removed to fit in EVP's byte-wise style
 Cipher = zuc-128-eea3
 Key = 173d14ba5003731d7a60049470f00a29
 IV = 6603549278
-Plaintext = 6cf65340735552ab0c9752fa6f9025fe0bd675d9005875b200
-Ciphertext = a6c85fc66afb8533aafc2518dfe784940ee1e4b030238cc800
+Plaintext = 6cf65340735552ab0c9752fa6f9025fe0bd675d9005875b2
+Ciphertext = a6c85fc66afb8533aafc2518dfe784940ee1e4b030238cc8
 
+# whole bytes test, no change
 Cipher = zuc-128-eea3
 Key = e5bd3ea0eb55ade866c6ac58bd54302a
 IV = 00056823c4
 Plaintext = 14a8ef693d678507bbe7270a7f67ff5006c3525b9807e467c4e56000ba338f5d429559036751822246c80d3b38f07f4be2d8ff5805f5132229bde93bbbdcaf382bf1ee972fbf9977bada8945847a2a6c9ad34a667554e04d1f7fa2c33241bd8f01ba220d
 Ciphertext = 131d43e0dea1be5c5a1bfd971d852cbf712d7b4f57961fea3208afa8bca433f456ad09c7417e58bc69cf8866d1353f74865e80781d202dfb3ecff7fcbc3b190fe82a204ed0e350fc0f6f2613b2f2bca6df5a473a57a4a00d985ebad880d6f23864a07b01
 
+# The trailing zero bytes are removed to fit in EVP's byte-wise style, and
+# some Ciphertext final trailing bits are changed to match the whole-byte
+# encryption
 Cipher = zuc-128-eea3
 Key = d4552a8fd6e61cc81a2009141a29c10b
 IV = 76452ec114
-Plaintext = 38f07f4be2d8ff5805f5132229bde93bbbdcaf382bf1ee972fbf9977bada8945847a2a6c9ad34a667554e04d1f7fa2c33241bd8f01ba220d3ca4ec41e074595f54ae2b454fd971432043601965cca85c2417ed6cbec3bada84fc8a579aea7837b0271177242a64dc0a9de71a8edee86ca3d47d033d6bf539804eca86c584a9052de46ad3fced65543bd90207372b27afb79234f5ff43ea870820e2c2b78a8aae61cce52a0515e348d196664a3456b182a07c406e4a20791271cfeda165d535ec5ea2d4df40000000
-Ciphertext = 8383b0229fcc0b9d2295ec41c977e9c2bb72e220378141f9c8318f3a270dfbcdee6411c2b3044f176dc6e00f8960f97afacd131ad6a3b49b16b7babcf2a509ebb16a75dcab14ff275dbeeea1a2b155f9d52c26452d0187c310a4ee55beaa78ab4024615ba9f5d5adc7728f73560671f013e5e550085d3291df7d5fecedded559641b6c2f585233bc71e9602bd2305855bbd25ffa7f17ecbc042daae38c1f57ad8e8ebd37346f71befdbb7432e0e0bb2cfc09bcd96570cb0c0c39df5e29294e82703a637f80000000
+Plaintext = 38f07f4be2d8ff5805f5132229bde93bbbdcaf382bf1ee972fbf9977bada8945847a2a6c9ad34a667554e04d1f7fa2c33241bd8f01ba220d3ca4ec41e074595f54ae2b454fd971432043601965cca85c2417ed6cbec3bada84fc8a579aea7837b0271177242a64dc0a9de71a8edee86ca3d47d033d6bf539804eca86c584a9052de46ad3fced65543bd90207372b27afb79234f5ff43ea870820e2c2b78a8aae61cce52a0515e348d196664a3456b182a07c406e4a20791271cfeda165d535ec5ea2d4df40
+Ciphertext = 8383b0229fcc0b9d2295ec41c977e9c2bb72e220378141f9c8318f3a270dfbcdee6411c2b3044f176dc6e00f8960f97afacd131ad6a3b49b16b7babcf2a509ebb16a75dcab14ff275dbeeea1a2b155f9d52c26452d0187c310a4ee55beaa78ab4024615ba9f5d5adc7728f73560671f013e5e550085d3291df7d5fecedded559641b6c2f585233bc71e9602bd2305855bbd25ffa7f17ecbc042daae38c1f57ad8e8ebd37346f71befdbb7432e0e0bb2cfc09bcd96570cb0c0c39df5e29294e82703a637f84
 
```

Comment:
This test is now *longer* (1576 bits) than the original (1570 bits). Suggest peeling off the last byte.

Reply:
Hmmmm, I haven't quite followed...  did you decide to shave off bits that are past the byte boundary?  Meaning that the shaved test will be 1568 bits?

Reply:
> Meaning that the shaved test will be 1568 bits?

Yes, IMO that is the reasonable solution. (Otherwise, what goes into the test is some extra bits that we essentially have no paper trail for.)

Reply:
Okay, I agree with this. I will shorten this along with other cases...